ws2812/gen-ws2812.go

```diff
@@ -252,7 +252,7 @@ func writeGoWrapper(f *os.File, arch string, megahertz int) error {
 	fmt.Fprintf(buf, "	portClear, maskClear := d.Pin.PortMaskClear()\n")
 	fmt.Fprintf(buf, "\n")
 	fmt.Fprintf(buf, "	mask := interrupt.Disable()\n")
-	fmt.Fprintf(buf, "	C.ws2812_writeByte%d(C.char(c), portSet, portClear, maskSet, maskClear)\n", megahertz)
+	fmt.Fprintf(buf, "	C.ws2812_writeByte%d(C.char(c), (*uint32)(unsafe.Pointer(portSet)), (*uint32)(unsafe.Pointer(portClear)), maskSet, maskClear)\n", megahertz)
```

Comment:
RP2040 is the odd one here, it returns a `*volatile.Register32` instead of a `*uint32` like all the others.
I don't remember the reasoning behind the `*uint32` API but I think it would be better to fix it in the machine package instead of working around it here.

Reply:
The volatile.Register32 seems to come straight from the SVD file of the rp2040. 
Here is a possible way to align https://github.com/tinygo-org/tinygo/pull/2872/files